docs/tutorials/elasticsearch.ipynb

```diff
@@ -0,0 +1,757 @@
+        "from elasticsearch import Elasticsearch\n",
+        "import numpy as np\n",
+        "import pandas as pd\n",
+        "import tensorflow as tf\n",
+        "from tensorflow.keras import layers\n",
+        "from tensorflow.keras.layers.experimental import preprocessing\n",
+        "import tensorflow_io as tfio"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "eCgO11GTJaTj"
+      },
+      "source": [
+        "### Validate tf and tfio imports"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "dX74RKfZ_TdF"
+      },
+      "outputs": [],
+      "source": [
+        "print(\"tensorflow-io version: {}\".format(tfio.__version__))\n",
+        "print(\"tensorflow version: {}\".format(tf.__version__))"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "yZmI7l_GykcW"
+      },
+      "source": [
+        "## Download and setup the Elasticsearch instance\n",
+        "\n",
+        "For demo purposes, the open-source version of the elasticsearch package is used.\n"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "YUj0878jPyz7"
+      },
+      "outputs": [],
+      "source": [
+        "!wget https://artifacts.elastic.co/downloads/elasticsearch/elasticsearch-oss-7.9.2-linux-x86_64.tar.gz\n",
+        "!wget https://artifacts.elastic.co/downloads/elasticsearch/elasticsearch-oss-7.9.2-linux-x86_64.tar.gz.sha512\n",
+        "!shasum -a 512 -c elasticsearch-oss-7.9.2-linux-x86_64.tar.gz.sha512 \n",
+        "!tar -xzf elasticsearch-oss-7.9.2-linux-x86_64.tar.gz\n",
+        "!chown -R daemon:daemon elasticsearch-7.9.2/\n"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "vAzfu_WiEs4F"
+      },
+      "source": [
+        "Run the instance as a daemon process"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "n9ujlunrWgRx"
+      },
+      "outputs": [],
+      "source": [
+        "es_server = Popen(['elasticsearch-7.9.2/bin/elasticsearch'], \n",
+        "                  stdout=PIPE, stderr=STDOUT,\n",
+        "                  preexec_fn=lambda: os.setuid(1))\n",
+        "\n",
+        "# Sleep for few seconds to let the instance start.\n",
+        "time.sleep(5)"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "f6qxCdypE1DD"
+      },
+      "source": [
+        "Once the instance has been started, grep for `elasticsearch` in the processes list to confirm the availability."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "48LqMJ1BEHm5"
+      },
+      "outputs": [],
+      "source": [
+        "!ps -ef | grep elasticsearch"
```

Comment:
While trying the notebook I noticed that elasticsearch may take several seconds to be ready (encountered `connection refused` error the first time. Maybe its worth to have a `wait for 10 seconds` to make sure server is ready?

Reply:
increased it to 15 seconds.